src/dune_console/progress.ml

```diff
@@ -3,25 +3,37 @@ open Stdune
 module No_flush = struct
   let status_line = ref Pp.nop
   let start () = ()
-  let status_line_len = ref 0
-
-  let hide_status_line () =
-    if !status_line_len > 0 then Printf.eprintf "\r%*s\r" !status_line_len ""
+  let status_line_height = ref 0
+
+  let hide_status_line =
+    (* This ansi code will clear the entire line leaving the cursor in place. *)
+    let ansi_erase = "\027[2K" in
+    (* Move the cursor up one line. *)
+    let ansi_up = "\027M" in
+    fun () ->
+      if !status_line_height > 0
+      then (
+        (* Return the cursor to the begining of the line and clear it. *)
+        Printf.eprintf "\r%s" ansi_erase;
+        for _ = 1 to !status_line_height - 1 do
+          (* If we have multiple lines, we move the cursor up and clear. *)
+          Printf.eprintf "%s%s" ansi_up ansi_erase
+        done)
   ;;
 
-  let show_status_line () = if !status_line_len > 0 then Ansi_color.prerr !status_line
+  let show_status_line () = if !status_line_height > 0 then Ansi_color.prerr !status_line
 
   let set_status_line = function
     | None ->
       hide_status_line ();
       status_line := Pp.nop;
-      status_line_len := 0
+      status_line_height := 0
     | Some line ->
       let line = Pp.map_tags line ~f:User_message.Print_config.default in
-      let line_len = String.length (Format.asprintf "%a" Pp.to_fmt line) in
       hide_status_line ();
       status_line := line;
-      status_line_len := line_len;
+      status_line_height
+        := Format.asprintf "%a" Pp.to_fmt line |> String.split_lines |> List.length;
```

Comment:
This is a guess and as such breaks when the terminal width causes the status line to wrap. I need to work out a better way to calculate how high the status line will be whilst still taking into account the width of the terminal.